ietf/ietfauth/views.py

```diff
@@ -491,9 +491,19 @@ def password_reset(request):
             if not user:
                 # try to find user ID from the email address
                 email = Email.objects.filter(address=submitted_username).first()
-                if email and email.person and email.person.user:
-                    user = email.person.user
-
+                if email and email.person:
+                    if email.person.user:
+                        user = email.person.user
+                    else: 
+                        # Create a User record with this (conditioned by way of Email) username
+                        # Don't bother setting the name or email fields on User - rely on the
+                        # Person pointer.
+                        user = User.objects.create(
+                            username=email.address.lower(), 
+                            is_active=True,
```

Comment:
Doesn't this lack the verification of the email though?

Reply:
To my read it's ok. It does create a `User` without verification, but it's a `User` that should already have existed and login won't be possible until they use the reset email which proves they received it.

(I haven't reviewed fully yet)

Reply:
That's correct  - the user's password will initially be something unusable. Only a successful response to the password reset email (which serves as proof of control of the email) will make the user object usable.